src/App.tsx

```diff
@@ -29,67 +30,126 @@ type ReorderOptions = {
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+    const {
+      LENGTH,
+      ALPHABET,
+      NONE,
+    } = SortType;
+    const isSorted = isReversed || (sortType !== NONE);
+    const setSortTypeState = (inputType: SortType) => (
+      this.setState({ sortType: inputType })
+    );
+    const setDefaultState = () => (
+      this.setState({
+        isReversed: false,
+        sortType: NONE,
+      })
+    );
+    const switchIsReversedState = () => (
+      this.setState((state) => (
+        { isReversed: !state.isReversed }
+      ))
+    );
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames('button is-info', {
+              'is-light': sortType !== ALPHABET,
+            })}
+            onClick={() => (setSortTypeState(ALPHABET))}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button is-success', {
+              'is-light': sortType !== LENGTH,
+            })}
+            onClick={() => (setSortTypeState(LENGTH))}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button is-warning', {
+              'is-light': !isReversed,
+            })}
+            onClick={switchIsReversedState}
+          >
+            Reverse
+          </button>
+
+          {isSorted && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={setDefaultState}
+            >
+              Reset
+            </button>
+          )}
+        </div>
 
-      <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          <ul>
+            {getReorderedGoods(
+              goodsFromServer,
+              this.state,
+            ).map((good, index) => (
```

Comment:
better to save result of `getReorderedGoods` to constant and then use here, it will be more understandable then using map right after function call 🙂